src/main/java/redis/clients/jedis/JedisSlotBasedConnectionHandler.java

```diff
@@ -61,7 +61,14 @@ public Jedis getConnectionFromSlot(int slot) {
       // assignment
       return connectionPool.getResource();
     } else {
-      return getConnection();
+      renewSlotCache(); //It's abnormal situation for cluster mode, that we have just nothing for slot, try to rediscover state
```

Comment:
It seems nice cause renewSlotCache() should succeed to renew slot cache if cluster is not down. Great improvement.